pkg/services/ngalert/prom/models.go

```diff
@@ -1,9 +1,19 @@
 package prom
 
 import (
+	"fmt"
+
 	prommodel "github.com/prometheus/common/model"
 )
 
+type ValidationError struct {
```

Comment:
I wonder if we can use `errutil.ErrorBase` here. It will give us structured errors, better logging and status codes.

Reply:
Good point, updated: https://github.com/grafana/grafana/pull/100558/commits/0eed3d632c45740a3ec7aca7eb15c7273576b2d5